homeassistant/components/binary_sensor/abode.py

```diff
@@ -17,38 +17,37 @@
 
 def setup_platform(hass, config, add_devices, discovery_info=None):
     """Set up a sensor for an Abode device."""
-    abode = hass.data[DATA_ABODE]
+    import abodepy.helpers.constants as CONST
+    import abodepy.helpers.timeline as TIMELINE
 
-    device_types = map_abode_device_class().keys()
+    data = hass.data[DOMAIN]
 
-    sensors = []
-    for sensor in abode.get_devices(type_filter=device_types):
-        sensors.append(AbodeBinarySensor(abode, sensor))
+    device_types = [CONST.TYPE_CONNECTIVITY, CONST.TYPE_MOISTURE,
+                    CONST.TYPE_MOTION, CONST.TYPE_OCCUPANCY,
+                    CONST.TYPE_OPENING]
 
-    add_devices(sensors)
+    devices = []
+    for device in data.abode.get_devices(generic_type=device_types):
+        if device.device_id not in data.exclude:
+            devices.append(AbodeBinarySensor(data, device))
 
+    for automation in data.abode.get_automations(
+            generic_type=CONST.TYPE_QUICK_ACTION):
+        if automation.automation_id not in data.exclude:
+            devices.append(AbodeQuickActionBinarySensor(
+                data, automation, TIMELINE.AUTOMATION_EDIT_GROUP))
 
-def map_abode_device_class():
-    """Map Abode device types to Home Assistant binary sensor class."""
-    import abodepy.helpers.constants as CONST
+    data.devices.extend(devices)
 
-    return {
-        CONST.DEVICE_GLASS_BREAK: 'connectivity',
-        CONST.DEVICE_KEYPAD: 'connectivity',
-        CONST.DEVICE_DOOR_CONTACT: 'opening',
-        CONST.DEVICE_STATUS_DISPLAY: 'connectivity',
-        CONST.DEVICE_MOTION_CAMERA: 'connectivity',
-        CONST.DEVICE_WATER_SENSOR: 'moisture'
-    }
+    add_devices(devices)
 
 
 class AbodeBinarySensor(AbodeDevice, BinarySensorDevice):
     """A binary sensor implementation for Abode device."""
 
-    def __init__(self, controller, device):
+    def __init__(self, data, device):
```

Comment:
You can remove the init here since you don't implement anything specific for this child class. The parent class init will catch the call arguments.